integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala

```diff
@@ -650,6 +650,51 @@ object GlobalDictionaryUtil extends Logging {
   }
 
   /**
+   * get file headers from fact file
+   *
+   * @param carbonLoadModel
+   * @return headers
+   */
+  private def getHeaderFormFactFile(carbonLoadModel: CarbonLoadModel): Array[String] = {
+    var headers: Array[String] = null
+    var factFile: String = null
+    val fileType = FileFactory.getFileType(carbonLoadModel.getFactFilePath)
+    val filePath = FileFactory.getCarbonFile(carbonLoadModel.getFactFilePath, fileType)
+    if (filePath.isDirectory) {
+      val listFiles = filePath.getParentFile.listFiles()
```

Comment:
@foryou2030 ....Please clarify the below points.
1. Why you are using filePath.getParentFile?
2. There can be a case where one folder can contain only subFolders. In that case you will not be find a csv file. Ideally to find a file it should be a recursive call.

@QiangCai ...please review this method and correct me if I am wrong